src/popup/plugins/i18n.ts

```diff
@@ -0,0 +1,69 @@
+import { computed } from 'vue';
+import { createI18n } from 'vue-i18n';
+import en from '@/popup/locales/en.json';
+
+export interface ILanguageConfig {
+  name: string;
+  getMessages: () => Promise<any>;
+}
+const ACTIVE_LANGUAGE = 'active-language';
+
+export const SUPPORTED_LANGUAGES = {
+  en: 'en',
+  cn: 'cn',
+} as const;
+
+export type SupportedLanguage = keyof typeof SUPPORTED_LANGUAGES;
+
+export const FALLBACK_LOCALE = SUPPORTED_LANGUAGES.en;
+
+export const i18n = createI18n({
+  allowComposition: true,
+  fallbackLocale: FALLBACK_LOCALE,
+  locale: FALLBACK_LOCALE,
+  messages: { en },
+});
+
+// @ts-ignore type coming from VueI18n is excessively deep and possibly infinite
+export const tg = i18n.global.t;
+
+export const languages: Record<SupportedLanguage, ILanguageConfig> = {
+  [SUPPORTED_LANGUAGES.en]: {
+    name: 'English',
+    getMessages: () => en,
+  },
+  [SUPPORTED_LANGUAGES.cn]: {
+    name: '中国人',
+    getMessages: () => import(/* webpackChunkName: "locale-cn" */ '../locales/cn.json'),
+  },
+};
+
+function getPreferredLanguage(): SupportedLanguage {
+  const [code] = navigator.language.split('-') as [SupportedLanguage];
+  return languages[code] ? code : FALLBACK_LOCALE;
+}
+
+export const activeLanguage = computed(
+  () => (i18n.global.locale as any).value || getPreferredLanguage(),
+);
```

Comment:
Is there actual need for using `getPreferredLanguage` as a fallback? I think if we set up the `locale` and `fallbackLocale` to `en` when using the `createI18n` the value should always be OK.

Also I'm not 100% sure, but maybe it would be better to do it this way:
```suggestion
export const activeLanguage = i18n.global.locale as SupportedLanguage;
```